Automattic/wp-calypso
# Plans 2024: Migrate pricing meta hook to Plans data-store

Related to https://github.com/Automattic/wp-calypso/issues/79005

## Proposed Changes

Migrates `usePricingMetaForGridPlans` to the Plans data-store and exports from the main index.
- Along with the relevant types
- Updated function docs

## Testing Instructions



* Ensure pricing works in `/start/plans` and `/plans/[site]`
* More general sanity checks with flows, WooExpress sites (showing intro offers)
    * It would be surprising though if breakage isn't caught by the build and tests

## Pre-merge Checklist
